vllm/config/compilation.py

```diff
@@ -137,6 +137,97 @@ def __post_init__(self) -> None:
                 )
 
 
+"""
+Note on dynamic shapes guard-dropping and reliability:
+=====================================================
+torch.compile() is designed to guard on dynamic shapes with no hesitation
+when needed. This contradicts with vLLM's torch.compile() approach of
+dropping the guards since many of those guards could be material.
+
+torch.compile() provides two kinds of dynamic shapes: backed and unbacked.
+torch.compile() guards on backed dynamic shapes and does not provide a
+guarantee that no guards will be added to them. User code, dynamo,
+inductor, and autograd all can add guards. Moreover, for 0/1
+specializations, backed symbols are specialized unconditionally to 0, 1,
+or >2 even without encountering a branching on those ranges.
+
+On the contrary, unbacked dynamic shapes are guaranteed not to be guarded
+on and are not 0/1 specialized. However, there is a possibility of
+throwing a data dependent error when a branch that requires their value is
+encountered and no explicit unbacked handling is defined. The framework is
+converging to a state where it won't throw DDE but rather pick general
+paths. One downside of using unbacked is missed optimization opportunities
+due to either perf bugs or picking general paths, also using a fixed
+non-example input-based hint (this will be fixed soon with override_hint
+API). An example of picking general paths is assuming input not contiguous
+in functions call contiguous() and reshape() when can't be symbolically proven
+with a change of introducing a clone.
+
+backed_size_oblivious is a flag that enables treating backed symbols as
+unbacked wherever explicit handling for unbacked is defined. With this
+mode, 0/1 specializations are mostly avoided in framework code and the
+default 0/1 specialization does not happen. However, there is still no
+guarantee that torch.compile won't guard, especially due to user code or
+custom passes. backed_size_oblivious is experimental in PyTorch compile
+and could be deprecated. That said, it's a safer option to use than
+backed and the probability of reducing performance is lower than
+unbacked.
+"""
+
+
+class DynamicShapesType(str, enum.Enum):
+    """Types of dynamic shapes handling in torch.compile()."""
+
+    BACKED = "backed"
+    """Use backed dynamic shapes. torch.compile() guards on backed dynamic
+    shapes and may add guards. Symbols are specialized to 0, 1, or >2 even
+    without encountering branching on those ranges."""
+
+    UNBACKED = "unbacked"
+    """Use unbacked dynamic shapes. Guaranteed not to be guarded on and not
+    0/1 specialized, but may throw data dependent errors when branches require
+    their value without explicit unbacked handling."""
+
+    BACKED_SIZE_OBLIVIOUS = "backed_size_oblivious"
+    """Experimental flag that treats backed symbols as unbacked when explicit
+    unbacked handling is defined."""
+
+
+@config
+@dataclass
+class DynamicShapesConfig:
```

Comment:
Why have a separate DynamicShapesConfig (instead of putting the fields into CompilationConfig)? Are you going to add significantly more fields?

Reply:
yeh I am going to add more stuff, 
eval_ds_guards
and 
error_at_inductor_specialization

are things in mind coming in next PR.